Clinical-Genomics/scout
# SV gene variantS search

This PR adds a functionality or fixes a bug.

- Fix #1179 - With mongodb 5.0 and higher it is possible to hold multiple partial indices on the same collection and keys, given that the partial criteria do not overlap. Use that to be able to set up a gene variantS search for the SV category as well. An initial test seemed decently quick, so maybe we will not need to filter on size, which is a bit complicated with inf and small sizes in the bag. 
- The hits may be a bit messy with large variants, so we will at least add a filter on category to the search page. A "length" filter could be considered as well; could be part of this PR or a later issue if it is needed.

- Backgrounding is not needed with the new mongodb 7.0 indexing logic. They do a blocking lock during the start and finish of the indexing, but otherwise runs a very fast, but still read-write indexing. Neato!

We prepared for this during the db migration, and have a category-less version of this index in place on prod.

The stage collection built quickly, opting for the same style as on prod, ie one index for all variant categories. rankscore >= 5 effectively excludes most of cancer, Mei variants, etc that are not yet ranked.
<img width="1053" alt="Screenshot 2024-06-28 at 16 05 07" src="https://github.com/Clinical-Genomics/scout/assets/758570/caea4a95-7172-4688-a14c-f16c15c6116e">

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>


**How to test**:
1. search for a few genes on stage, note both SVs and SNVs in results
2. toggle search to only SNV or only SV and see that the filter behaves

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
